Sources/Datasets/Dataset.swift

```diff
@@ -0,0 +1,6 @@
+protocol S5TFDataset {
+    associatedtype _Iterator // swiftlint:disable:this type_name
+    func batched(_ batchSize: Int) -> _Iterator
+}
+
+protocol S5TFDatasetIterator: Sequence, IteratorProtocol {}
```

Comment:
Would be cool if we could specify that the underlying `dataset` should conform to `S5TFDataset`, is that possible?

Reply:
Good idea!